packages/amplify-graphql-transformer-core/src/transformation/validation.ts

```diff
@@ -142,3 +145,21 @@ export const validateModelSchema = (doc: DocumentNode) => {
   const schema = buildASTSchema(fullDocument);
   return validate(schema, fullDocument, specifiedRules);
 };
+
+export const validateAuthModes = (authConfig: AppSyncAuthConfiguration) => {
+  let additionalAuthModes: AppSyncAuthMode[] = [];
+
+  if (authConfig.additionalAuthenticationProviders) {
+    additionalAuthModes = authConfig.additionalAuthenticationProviders.map(p => p.authenticationType).filter(t => !!t);
```

Comment:
The `map()` isn't necessary here. The `filter()` call can just check `p.authenticationType`.

Reply:
Synced offline - keeping as is since it's reducing the object as well.